code/modules/client/preferences.dm

```diff
@@ -542,7 +542,8 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 /// Sanitizes the preferences, applies the randomization prefs, and then applies the preference to the human mob.
 /datum/preferences/proc/safe_transfer_prefs_to(mob/living/carbon/human/character, icon_updates = TRUE, is_antag = FALSE)
 	apply_character_randomization_prefs(is_antag)
-	apply_prefs_to(character, icon_updates)
+	spawn(1) /// SKYRAT EDIT: delaying this slightly to resolve some wacky timing bugs that eat charprefs
+		apply_prefs_to(character, icon_updates)
```

Comment:
Would you mind giving something like `IMMEDIATE_INVOKE_ASYNC()` (or whatever it's actually spelled like) a try? I'd rather avoid using `spawn()` if we can, it's something we try and avoid as much as possible in the codebase, for debugging reasons.

Reply:
No prob, I didn't know spawn() had debug issues!

Reply:
ADDENDUM: I tried it with INVOKE_ASYNC and it broke in *new* ways.

Reply:
Put in a little wrapper function that sleeps to restore the expected behaviour while still using INVOKE_ASYNC.  It's a wee bit ugly but it works as expected at least.